packages/button/src/Button/components/GenericButton.spec.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react"
+import { render, screen } from "@testing-library/react"
+import GenericButton, { CustomButtonProps } from "./GenericButton"
+
+it("renders a custom component when the 'component' prop is provided", () => {
+  render(<ExampleComponent />)
+  expect(
+    screen.getByRole("button", { name: "A custom button" })
+  ).toBeInTheDocument()
+})
+
+function ExampleComponent() {
```

Comment:
I think declarations should be made before usage. It's a lot clearer, as opposed to needing to potentially search throughout the file to find it.

I'm also a little unsure about putting the ref into the test and then not actually testing it (I know we shouldn't be testing internal behaviour, so no issues there). It gives a slight impression that it must be declared when using the component.

Reply:
Gotta say I disagree, I think we should be doing this much more often. IMO the main purpose/function of the file should be at the top, and any supporting private functions should be a secondary focus below that.

Reply:
There's actually linting rules that exist for it, as it is a more common practice for reducing confusion (I'm one of them who finds it confusing). It would be much easier to enforce moving forward too
https://eslint.org/docs/latest/rules/no-use-before-define

>In JavaScript, prior to ES6, variable and function declarations are hoisted to the top of a scope, so it’s possible to use identifiers before their formal declarations in code. **This can be confusing and some believe it is best to always declare variables and functions before using them.**